### DISCUSSION TITLE
far field of the metalens

### DISCUSSION THREAD

**Discussion by Diego-Miao** [facebookresearch/fmmax#111] [Q&A]
Hi,

I am a graduated student from Taiwan.

Currently, I am going to develop a design flow for large-area metalens.

I plan to decompose the metalens and then optimize each sub-metalens with aperiodic Fourier modal method.

It is mentioned by this jornal:
https://www.nature.com/articles/s41377-019-0159-5

But, how can I get the far field of each sub-metalens?

Thanks for your help.

**Comment by mfschubert** [comment]
Hi @Diego-Miao , this sounds like an interesting application. However, the far field of an individual unit cell seems to be a somewhat poorly defined thing. Can you be a bit more precise about what you are looking to compute?

Perhaps you are looking for a nearfield to farfield transformation? E.g. the inputs are the fields (Ex, Ey, Ez, Hx, Hy, Hz) on a grid in an xy plane, and the outputs are the fields at one or more arbitrary xyz points? If so, this is not something that fmmax currently provides, but it would be a useful addition.

**Reply by Diego-Miao** [reply]
Hi @mfschubert,
Yes, I am looking for a nearfield to farfield transformation.
I wonder if it is far field conversion, which is Rayleigh-Sommerfeld Diffraction Integral.

Thanks.

**Reply by mfschubert** [reply]
Yes, this or something like the angular spectrum method may be appropriate here. We don't presently have a plan to implement this, but we would consider a contribution if you are interested.
